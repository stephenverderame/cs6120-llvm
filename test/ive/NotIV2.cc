// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    int j = 0;
    for (int i = 5000; i >= m; i -= m, ++j) {
        int p = j + i;
        printf("%d\n", p);
    }
}
// NOLINTEND