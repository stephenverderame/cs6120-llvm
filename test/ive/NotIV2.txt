5000
4981
4962
4943
4924
4905
4886
4867
4848
4829
4810
4791
4772
4753
4734
4715
4696
4677
4658
4639
4620
4601
4582
4563
4544
4525
4506
4487
4468
4449
4430
4411
4392
4373
4354
4335
4316
4297
4278
4259
4240
4221
4202
4183
4164
4145
4126
4107
4088
4069
4050
4031
4012
3993
3974
3955
3936
3917
3898
3879
3860
3841
3822
3803
3784
3765
3746
3727
3708
3689
3670
3651
3632
3613
3594
3575
3556
3537
3518
3499
3480
3461
3442
3423
3404
3385
3366
3347
3328
3309
3290
3271
3252
3233
3214
3195
3176
3157
3138
3119
3100
3081
3062
3043
3024
3005
2986
2967
2948
2929
2910
2891
2872
2853
2834
2815
2796
2777
2758
2739
2720
2701
2682
2663
2644
2625
2606
2587
2568
2549
2530
2511
2492
2473
2454
2435
2416
2397
2378
2359
2340
2321
2302
2283
2264
2245
2226
2207
2188
2169
2150
2131
2112
2093
2074
2055
2036
2017
1998
1979
1960
1941
1922
1903
1884
1865
1846
1827
1808
1789
1770
1751
1732
1713
1694
1675
1656
1637
1618
1599
1580
1561
1542
1523
1504
1485
1466
1447
1428
1409
1390
1371
1352
1333
1314
1295
1276
1257
1238
1219
1200
1181
1162
1143
1124
1105
1086
1067
1048
1029
1010
991
972
953
934
915
896
877
858
839
820
801
782
763
744
725
706
687
668
649
630
611
592
573
554
535
516
497
478
459
440
421
402
383
364
345
326
307
288
269
