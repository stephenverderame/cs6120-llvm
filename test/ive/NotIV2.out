Basic IV:   %3 = phi i32 [ 0, %0 ], [ %8, %2 ]
Basic IV:   %4 = phi i32 [ 5000, %0 ], [ %7, %2 ]
Basic IV:   %7 = add nsw i32 %4, -20
Basic IV:   %8 = add nuw nsw i32 %3, 1
